### ISSUE TITLE
MySQL `upsert()`: `Internal error: Attempted to serialize empty result.`

### ISSUE THREAD

**Issue by bastianparedes** [prisma/prisma#17982]
Hi Prisma Team! My Prisma Client just crashed. This is the report:
## Versions

| Name            | Version            |
|-----------------|--------------------|
| Node            | v19.5.0            | 
| OS              | darwin-arm64       |
| Prisma Client   | 4.10.1             |
| Query Engine    | aead147aa326ccb985dcfed5b065b4fdabd44b19|
| Database        | mysql              |

## Logs
```
prisma:client:libraryEngine sending request, this.libraryStarted: false
prisma:client:libraryEngine library starting
prisma:client:libraryEngine library started
```

## Client Snippet
```ts
const result = await prisma.campaign.upsert({
      create: {
        idStatus: campaign.idStatus,
        name: campaign.name
      },
      update: {
        idStatus: campaign.idStatus,
        name: campaign.name
      },
      where: {
        idCampaign: campaign.idCampaign
      }
    });
```

## Schema
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider     = "mysql"
  url          = env("DATABASE_URL")
  relationMode = "prisma"
}

model campaign {
  idCampaign Int         @id @default(autoincrement())
  name        String      @default("") @db.VarChar(45)
  idStatus   Int         @default(0)
  status      status      @relation(fields: [idStatus], references: [idStatus], onDelete: NoAction, onUpdate: NoAction, map: "status_ibfk_1")
  evaluator   evaluator[]
  variation   variation[]

  @@index([idStatus], map: "status_ibfk_1")
}

model status {
  idStatus Int        @id @unique(map: "idStatus_UNIQUE")
  value     String     @unique(map: "value_UNIQUE") @db.VarChar(45)
  campaign  campaign[]
}

model evaluator {
  idEvaluator Int
  idCampaign  Int
  name         String   @default("") @db.VarChar(45)
  javascript   String   @default("") @db.VarChar(5000)
  campaign     campaign @relation(fields: [idCampaign], references: [idCampaign], onDelete: NoAction, onUpdate: NoAction, map: "evaluator_ibfk_1")

  @@id([idEvaluator, idCampaign])
  @@index([idCampaign], map: "evaluator_ibfk_1")
}

model variation {
  idVariation Int
  idCampaign  Int
  name         String   @default("") @db.VarChar(45)
  html         String   @default("") @db.VarChar(5000)
  css          String   @default("") @db.VarChar(5000)
  javascript   String   @default("") @db.VarChar(5000)
  traffic      Int      @default(0)
  campaign     campaign @relation(fields: [idCampaign], references: [idCampaign], onDelete: NoAction, onUpdate: NoAction, map: "variation_ibfk_1")

  @@id([idVariation, idCampaign])
  @@index([idCampaign], map: "variation_ibfk_1")
}

```

## Prisma Engine Query
```
{"X":{}}
```

Also I want to add: the problem only happens when I want to update in this upsert.

**Comment by bastianparedes** [comment]
When I want to upsert, it works the create part, when it panics when update.

**Comment by laplab** [comment]
Thank you for reporting the issue, @bastianparedes! We are currently testing the fix for this problem. Would it be possible for you to check your code against this version of Prisma?
```
npm i prisma@5.13.0-dev.30
```
Bear in mind that this is a development version and should not be used in production. This is only to check if the issue is fixed.

**Comment by laplab** [comment]
This issue was fixed in [Prisma 5.13.0](https://github.com/prisma/prisma/releases/tag/5.13.0).
